### PR TITLE
Use api route prefix parameter instead of `/api/v2`

### LINKED ISSUE #13854: Warning: Undefined array key "/api/v2/shop/product-images/{id}"
**Sylius version affected**: 1.11.3

**Description**  
We are steal using the old `/new-api` to preserve some legacy API calls, while loading the page `/new-api` on a web browser to get the swagger page this error is displayed :

```
ErrorException:
Warning: Undefined array key "/api/v2/shop/product-images/{id}"

  at vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/ProductImageDocumentationNormalizer.php:48
  at Sylius\Bundle\ApiBundle\Swagger\ProductImageDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/ProductSlugDocumentationNormalizer.php:34)
  at Sylius\Bundle\ApiBundle\Swagger\ProductSlugDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/ProductVariantDocumentationNormalizer.php:32)
  at Sylius\Bundle\ApiBundle\Swagger\ProductVariantDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/ShippingMethodDocumentationNormalizer.php:32)
  at Sylius\Bundle\ApiBundle\Swagger\ShippingMethodDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/PathHiderDocumentationNormalizer.php:33)
  at Sylius\Bundle\ApiBundle\Swagger\PathHiderDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Swagger/ShopAuthenticationTokenDocumentationNormalizer.php:35)
  at Sylius\Bundle\ApiBundle\Swagger\ShopAuthenticationTokenDocumentationNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/api-platform/core/src/Swagger/Serializer/ApiGatewayNormalizer.php:53)
  at ApiPlatform\Core\Swagger\Serializer\ApiGatewayNormalizer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/symfony/serializer/Serializer.php:161)
  at Symfony\Component\Serializer\Serializer->normalize(object(Documentation), 'json', array('spec_version' => 3))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/ApiPlatform/Bridge/Symfony/Bundle/Action/SwaggerUiAction.php:126)
  at Sylius\Bundle\ApiBundle\ApiPlatform\Bridge\Symfony\Bundle\Action\SwaggerUiAction->getContext(object(Request), object(Documentation))
     (vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/ApiPlatform/Bridge/Symfony/Bundle/Action/SwaggerUiAction.php:100)
  at Sylius\Bundle\ApiBundle\ApiPlatform\Bridge\Symfony\Bundle\Action\SwaggerUiAction->__invoke(object(Request))
     (vendor/symfony/http-kernel/HttpKernel.php:152)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:25) 
```

**Steps to reproduce**  
Change the parameter value of `sylius.security.new_api_route` to a different value like `/new-api` or `/my-api`

**Possible Solution**  

1. Use the parameter `%sylius.security.new_api_route%` as a service argument for the `Sylius\Bundle\ApiBundle\Swagger\ProductImageDocumentationNormalizer` service to build the right `SHOP_ITEM_PATH`.
2. This must be done also with the service `Sylius\Bundle\ApiBundle\Swagger\ProductSlugDocumentationNormalizer`.
3. Finally the `src/Sylius/Bundle/ApiBundle/Resources/config/app/config.yaml` line 8 to 10 must use the `%sylius.security.new_api_route%` prefix inside these strings.

### PR DESCRIPTION
**Pull request by Prometee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #13854
| License         | MIT

Small PR to be able to use another api route prefix while generating the swagger page.

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]
We should perhaps introduce some specs for these classes

File: src/Sylius/Bundle/ApiBundle/Swagger/ProductImageDocumentationNormalizer.php [hunk 1]
```diff
@@ -45,17 +38,18 @@ public function normalize($object, $format = null, array $context = [])
         $enums = $this->filterProvider->provideShopFilters();
         $enums = array_keys($enums);
 
-        $params = $docs['paths'][self::SHOP_ITEM_PATH]['get']['parameters'];
+        $shopProductImagePath = $this->apiRoute . '/shop/product-images/{id}';
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Swagger/ProductImageDocumentationNormalizer.php:41]: I would leave suffix as a const

**Comment by lchrusciel** [comment]
Thanks, Francis! :tada:

**Comment by Prometee** [comment]
@lchrusciel I'm also thinking about doing this kind of api route prefix for the Behat tests, because it forces us to use `/api/v2` too.

File: src/Sylius/Bundle/ApiBundle/Swagger/ProductImageDocumentationNormalizer.php (hunk 1 above)
Comment (Prometee) [reply src/Sylius/Bundle/ApiBundle/Swagger/ProductImageDocumentationNormalizer.php:41]: I also think of it but it was a private const so it wasn't editable nor accessible.

**Pull request merged**